### ISSUE TITLE
-Astubs doesn't write type arguments for outer class when inner class present

### ISSUE THREAD

**Issue by jwaataja** [typetools/checker-framework#3438]
When using `-Ainfer=stubs`, creating stub files for classes with inner classes fails to include type arguments in the outer class. Consider this class
```java
public class GenericClass<T> {
    public class InnerClass {
        // Create error to force generation of stub file, ignore.
        Object o = null;
    }
}
```
I compiled it with,
```bash
$CHECKERFRAMEWORK/checker/bin/javac -processor NullnessChecker -Ainfer=stubs GenericClass.java
```
When I inspect the generated stub file, I see
```java
import org.checkerframework.checker.initialization.qual.Initialized;
import org.checkerframework.checker.nullness.qual.Nullable;

class GenericClass {

  class InnerClass {

    // fields:

    @Initialized @Nullable java.lang.Object o;

  }
}
```
Notice that `GenericClass` has no type parameter.

When I try to use this stub file, I get an error:
```bash
$CHECKERFRAMEWORK/checker/bin/javac -processor NullnessChecker -Astubs=build GenericClass.java
```
```
warning: StubParser: annotateTypeParameters: mismatched sizes:  typeParameters (size 0)=[];  typeArguments (size 1)=[T extends Object];  decl=class GenericClass {      class InnerClass {          @Initialized         @Nullable         java.lang.Object o;     } };  elt=GenericClass (class com.sun.tools.javac.code.Symbol$ClassSymbol).; for more details, run with -AstubDebug
GenericClass.java:4: error: [assignment.type.incompatible] incompatible types in assignment.
        Object o = null;
                   ^
  found   : null
  required: @Initialized @NonNull Object
1 error
1 warning
```

Output for
```bash
$CHECKERFRAMEWORK/checker/bin/javac -processor NullnessChecker -Ainfer=stubs GenericClass.java -version -verbose -AprintAllQualifiers
```
[javac_create_stubs_output.txt](https://github.com/typetools/checker-framework/files/4881794/javac_create_stubs_output.txt)

Output for
```bash
$CHECKERFRAMEWORK/checker/bin/javac -processor NullnessChecker -Astubs=build GenericClass.java -version -verbose -AprintAllQualifiers
```

[javac_use_stubs_output.txt](https://github.com/typetools/checker-framework/files/4881795/javac_use_stubs_output.txt)
